examples/example-with-swagger.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+
+const fastify = require('fastify')({ logger: true })
+const Ajv = require('ajv')
+
+const ajv = new Ajv({
+  /**
+   * default values of Fastify
+   * Docs: https://www.fastify.io/docs/latest/Reference/Validation-and-Serialization/#validator-compiler
+   *   */
+
+  coerceTypes: 'array', // change data type of data to match type keyword
+  useDefaults: true, // replace missing properties and items with the values from corresponding default keyword
+  removeAdditional: true, // remove additional properties
+  uriResolver: require('fast-uri'),
+  addUsedSchema: false,
+  // Explicitly set allErrors to `false`.
+  // When set to `true`, a DoS attack is possible.
+  allErrors: false
+})
+ajv.addKeyword({
+  keyword: 'isFile',
+  compile: (_schema, parent, _it) => {
+    parent.type = 'file'
+    delete parent.isFile
+    return () => true
+  }
+})
+fastify.setValidatorCompiler(({ schema, method, url, httpPart }) => {
+  return ajv.compile(schema)
+})
```

Comment:
If you go this route, then any additional schemas will need to be added to the validator instead of the normal `fastify.addSchema` usage, which may not be intuitive. See [the note in the validator compiler docs](https://www.fastify.io/docs/latest/Reference/Validation-and-Serialization/#validator-compiler:~:text=Note%3A%20If%20you%20use%20a%20custom%20instance%20of%20any%20validator%20(even%20Ajv)%2C%20you%20have%20to%20add%20schemas%20to%20the%20validator%20instead%20of%20Fastify%2C%20since%20Fastify%27s%20default%20validator%20is%20no%20longer%20used%2C%20and%20Fastify%27s%20addSchema%20method%20has%20no%20idea%20what%20validator%20you%20are%20using.).

Instead of defining an entirely custom validator, would passing the custom keyword as options when initializing fastify allow users to still rely on Fastify's private AJV implementation?

- See [Fastify's AJV factory docs](https://www.fastify.io/docs/latest/Reference/Server/#factory-ajv)
- See the [AJV keywords option](https://ajv.js.org/options.html#keywords)

Reply:
you made a good point 
as myself I didn't use `addSchema()` but it's valid I will push changes